crates/client/eth/src/client.rs

```diff
@@ -0,0 +1,172 @@
+use crate::config::{L1StateUpdate, LogStateUpdate};
+use crate::utils::{convert_log_state_update, trim_hash, u256_to_starkfelt, LOG_STATE_UPDTATE_TOPIC};
+use alloy::consensus::TypedTransaction;
+use alloy::hex::decode;
+use alloy::primitives::{Address, BlockNumber, B256, U64};
+use alloy::providers::{Provider, ProviderBuilder, ReqwestProvider};
+use alloy::rpc::types::{Filter, FilterSet, TransactionRequest};
+use alloy::sol;
+use alloy::sol_types::SolEvent;
+use anyhow::{bail, Context};
+use bitvec::macros::internal::funty::Fundamental;
+use dc_db::DeoxysBackend;
+use std::sync::Arc;
+
+use dc_sync::metrics::block_metrics::BlockMetrics;
+use dp_convert::{ToFelt, ToStarkFelt};
+use dp_transactions::TEST_CHAIN_ID;
+use dp_utils::channel_wait_or_graceful_shutdown;
+use futures::StreamExt;
+use primitive_types::H256;
+use serde_json::Value;
+use starknet_api::hash::StarkFelt;
+use starknet_types_core::felt::Felt;
+use url::Url;
+
+pub struct EthereumClient {
+    provider: Arc<ReqwestProvider>,
+    l1_core_address: Address,
+}
+
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    StarknetCore,
+    "src/abis/starknet_core.json"
+);
+
+impl EthereumClient {
```

Comment:
we should test all these functions with integration tests using `anvil`

Reply:
tests added